hack/packages/templates/repo-utils/package-helpers.lib.yaml

```diff
@@ -1,17 +1,27 @@
 #@ load("@ytt:data", "data")
 #@ load("@ytt:assert", "assert")
 
-#@ def get_package_repository(repository_name):
-#@   return data.values.repositories[repository_name]
+#@ def get_package_repository(repository_name, package_name):
```

Comment:
If `data.values.packageRepository` is non-empty and matches [management](https://github.com/vmware-tanzu/tanzu-framework/blob/d11a2a745759e214571f1d60b62f63463f4c7182/packages/package-values.yaml#L4) or [standalone](https://github.com/vmware-tanzu/tanzu-framework/blob/d11a2a745759e214571f1d60b62f63463f4c7182/packages/package-values.yaml#L148), it returns a non-empty dictionary. Otherwise, it should return an empty dictionary.

Reply:
Thank you @danniel1205 for this intriguing question. Finding the answers has helped me to explore and advance my understanding of YTT greatly. Before this question, I didn't even know YTT had functions, nor did I ever hear about Starlark and that it's based on Python.

The answer to your question is, no. Calling the function with an empty string, `get_package_repository(data.values.packageRepository, "")`, will not return an empty string. The packageRepository is a required value to be passed in at this time and must exist in the `package-values.yaml` file. Currently, the package name must be empty . However, the option to specify a package name has been added in PR #1782.

As the package name will be empty, all the packages in the provided `package_repository` will be returned and will be in the output.

As I mentioned, the ability to specify a package name during `make package-repo-bundle` is not currently allowed. @yharish991 has allowed for the specification of a `package_name` because of PR #1782 which I thought would be merged before this PR.

We had to prioritize this PR over PR #1782, but I have retained the ability to specify a package name for this PR to minimize the number of changes going forward.

You might have asked, "Will it return empty string?" because of a playground experiment you did. I have tried to explain why that happened in [a different comment](https://github.com/vmware-tanzu/tanzu-framework/pull/2199#issuecomment-1109116682). Hopefully, that helps.